admin/abstract/views.py

```diff
@@ -0,0 +1,49 @@
+from django.views.generic import FormView
+from django.http import HttpResponseNotFound
+from django.shortcuts import render
+
+
+class GuidFormView(FormView):
+    form_class = None
+    template_name = None
+    object_type = None
+
+    def get_guid_object(self):
+        raise NotImplementedError
+
+    def __init__(self):
+        self.guid = None
+        super(GuidFormView, self).__init__()
+
+    def get(self, request, *args, **kwargs):
+        self.guid = request.GET.get('guid', None)
+        if self.guid is not None:
+            try:
+                guid_object = self.get_guid_object()
+            except (AttributeError, TypeError):
+                return HttpResponseNotFound(
```

Comment:
You on the right branch? I get the following:
<img width="420" alt="screen shot 2016-02-01 at 12 59 44 pm" src="https://cloud.githubusercontent.com/assets/3709/12726100/c4b86818-c8e3-11e5-89c3-45ad237a6d2a.png">